test/unit/context_unit_test.rb

```diff
@@ -468,11 +468,79 @@ def test_apply_global_filter
     assert_equal 'hi filtered', context.apply_global_filter('hi')
   end
 
+  def test_static_environments_are_read_with_lower_priority_than_environments
+    context = Context.build(
+      static_environments: { 'shadowed' => 'static', 'unshadowed' => 'static' },
+      environments: { 'shadowed' => 'dynamic' }
+    )
+
+    assert_equal 'dynamic', context['shadowed']
+    assert_equal 'static', context['unshadowed']
+  end
```

Comment:
I'm not sure why the Liquid-C tests didn't fail here. Liquid-C implements Context#find_variable, and doesn't handle static_environments. This is critical to look into before shipping this.

Reply:
This isn't failing as its looking at your https://github.com/Shopify/liquid-c/pull/43 which I think was forked before the new `Context#find_variable` changes.

Reply:
This is the reason I do like the idea of https://github.com/Shopify/liquid/issues/1134